politeiawww/cmd/piwww/editproposal.go

```diff
@@ -107,22 +121,55 @@ func (cmd *EditProposalCmd) Execute(args []string) error {
 
 		files = append(files, f)
 	}
-
 	// Setup metadata
-	if cmd.Name == "" {
-		r, err := util.Random(v1.PolicyMinProposalNameLength)
+	pm := v1.ProposalMetadata{}
+
+	// Use existing metadata flag is true
+	// fetch record and pre-fill data
+	if cmd.Usemd {
+		pdr, err := client.ProposalDetails(cmd.Args.Token,
+			&v1.ProposalsDetails{})
 		if err != nil {
 			return err
 		}
-		cmd.Name = hex.EncodeToString(r)
+		pm.Name = pdr.Proposal.Name
+		if pdr.Proposal.LinkTo != "" {
```

Comment:
All of these checks are unnecessary and add noise to the code.  Just set the proposal metadata to the existing proposal metadata object.